### ISSUE TITLE
File content encoding on diff view and patch view

### ISSUE THREAD

**Issue by futatuki** [viewvc/viewvc#301]
In case the encoding of the file content in the repository is not UTF-8,  I think the revision diff of those files should be transcoded to UTF-8 in diff view and shoud not be transcoded (say, as is) in patch view.  Is it correct?

If it is so, both in current HEAD of master (90e5e1c2b7d430805e036280ae8a2224af83afd5 at 2022-11-01T08:50:00Z) and master with PR #299 applied,  they had still trouble about it.

I started to fix it based on PR #299 branch.

**Comment by cmpilato** [comment]
That's correct.  The patch, we assume, is meant to be downloaded and applied to a source code working copy (which would have the matching encoding).  Moreover, because the patch content isn't HTML, we don't run the risk of mixing encodings like we for our decorated diff view.
